### ISSUE TITLE
strange artefacts when playing 4K Bluray

### ISSUE THREAD

**Issue by philcerf** [mpv-player/mpv#12700]
### Important Information

Provide following Information:
- mpv: 0.36.0
- Linux Distribution and Version: Debian unstable, kernel 6.5.6
- Source of the mpv binary: from debian
- Window Manager and version: Cinnamon 5.8.4
- GPU driver and version: Intel(R) Graphics (ADL GT2) (0x46a6), mesa 23.2.1, intel-media-va-driver 23.3.4

### Reproduction steps

When playing back a 4K bluray, I see strange artefacts like these:
![mpv-shot0001](https://github.com/mpv-player/mpv/assets/41550983/b385dc68-1f26-4347-b000-467a2cbaa861)
![mpv-shot0004](https://github.com/mpv-player/mpv/assets/41550983/e55e78f1-1083-48db-8703-fd64666767d1)
Sometimes it's the whole screen, sometimes only parts of it, sometimes (for a few frames) the image is fine.

The same problem doesn't happen when e.g. playing back with `ffplay`, though playback with that is unspeakably slow (it doesn't really play, more freeze and then jump a bit).

Playback with `mpv` is also pretty slow (to slow to be useful actually), but not as bad as with `ffplay`.

I do have `hwdec=auto` in my `mpv.conf` and would have expected that my CPU/GPU are recent enough to play back HVEC:
```
$ vainfo 
libva info: VA-API version 1.20.0
libva info: Trying to open /usr/lib/x86_64-linux-gnu/dri/iHD_drv_video.so
libva info: Found init function __vaDriverInit_1_20
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.20 (libva 2.12.0)
vainfo: Driver version: Intel iHD driver for Intel(R) Gen Graphics - 23.3.4 ()
vainfo: Supported profile and entrypoints
      VAProfileNone                   :	VAEntrypointVideoProc
      VAProfileNone                   :	VAEntrypointStats
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSliceLP
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSliceLP
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileJPEGBaseline           :	VAEntrypointEncPicture
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSliceLP
      VAProfileVP8Version0_3          :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointEncSliceLP
      VAProfileHEVCMain10             :	VAEntrypointVLD
      VAProfileHEVCMain10             :	VAEntrypointEncSliceLP
      VAProfileVP9Profile0            :	VAEntrypointVLD
      VAProfileVP9Profile0            :	VAEntrypointEncSliceLP
      VAProfileVP9Profile1            :	VAEntrypointVLD
      VAProfileVP9Profile1            :	VAEntrypointEncSliceLP
      VAProfileVP9Profile2            :	VAEntrypointVLD
      VAProfileVP9Profile2            :	VAEntrypointEncSliceLP
      VAProfileVP9Profile3            :	VAEntrypointVLD
      VAProfileVP9Profile3            :	VAEntrypointEncSliceLP
      VAProfileHEVCMain12             :	VAEntrypointVLD
      VAProfileHEVCMain422_10         :	VAEntrypointVLD
      VAProfileHEVCMain422_12         :	VAEntrypointVLD
      VAProfileHEVCMain444            :	VAEntrypointVLD
      VAProfileHEVCMain444            :	VAEntrypointEncSliceLP
      VAProfileHEVCMain444_10         :	VAEntrypointVLD
      VAProfileHEVCMain444_10         :	VAEntrypointEncSliceLP
      VAProfileHEVCMain444_12         :	VAEntrypointVLD
      VAProfileHEVCSccMain            :	VAEntrypointVLD
      VAProfileHEVCSccMain            :	VAEntrypointEncSliceLP
      VAProfileHEVCSccMain10          :	VAEntrypointVLD
      VAProfileHEVCSccMain10          :	VAEntrypointEncSliceLP
      VAProfileHEVCSccMain444         :	VAEntrypointVLD
      VAProfileHEVCSccMain444         :	VAEntrypointEncSliceLP
      VAProfileAV1Profile0            :	VAEntrypointVLD
      VAProfileHEVCSccMain444_10      :	VAEntrypointVLD
      VAProfileHEVCSccMain444_10      :	VAEntrypointEncSliceLP
```

Anyway, this graphics artefacts also shop up when disabling the hardware acceleration.

### Expected behavior

Playing back without any graphics artefacts ;-)

### Log file

Do you need a logfile from the whole playback?

### Sample files

That could be difficult. a) and 4K BluRay is huge b) there are copyright issues.

**Comment by Dudemanguy** [comment]
>I do have hwdec=auto in my mpv.conf

Assuming this works with `--hwdec=no` then it would probably be a mesa bug.

**Comment by philcerf** [comment]
Quoing myself:
> Anyway, this graphics artefacts also shop up when disabling the hardware acceleration.

**Comment by Dudemanguy** [comment]
Definitely a driver issue then.

**Comment by philcerf** [comment]
But why does it then not happen with `ffplay`?

**Comment by Dudemanguy** [comment]
Because you weren't using hardware decoding with ffplay?

**Comment by philcerf** [comment]
Okay I can try it again with `ffplay` and with hardware decoding (once I found out how to do that ^^ … you don't happen to know?).

But still if `mpv` also **shows** the same issues with **no** hardware decoding... how should it be the driver, if `ffplay` does **not** show them without hardware decoding?

**Comment by Dudemanguy** [comment]
Oh wait, I'm sorry I'm totally confused. I thought you said the artifacts didn't show up with hardware decoding for some reason... No clue then what is happening here.

**Comment by philcerf** [comment]
Well... anything I could help in debugging this?

**Comment by Dudemanguy** [comment]
Sure, identify what kind of codecs this happens in, see if it happens in all graphics apis or just some, see if you can bisect to an mpv that causes this, or see if it's a particular mesa version, provide a sample file, etc. Can't do much off what we have so far

**Comment by philcerf** [comment]
First, I just tried with vlc 3.0.19, and there playback seems to work find, and even in decent speed.
I really never use `vlc`, but it seems it would auto-use VAAPI (there's some vaapi debug output in the terminal).

Well as said, providing a sample file might be problematic... not only because of the size of a 4K bluray, but also because of legal issues.

And again, given that it works in `ffplay` and in `vlc` even with VAAPI... I don't quite see where I should start with trying different mesa versions.

It's also no regression, not that I'd know at least, so what should I bisect?

As for the codecs, `ffprobe` says:
```
$ ffprobe bluray:. 
ffprobe version 6.0-7+b1 Copyright (c) 2007-2023 the FFmpeg developers
  built with gcc 13 (Debian 13.2.0-4)
  configuration: --prefix=/usr --extra-version=7+b1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-sndio --enable-libjxl --enable-pocketsphinx --enable-librsvg --enable-libvpl --disable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --disable-libsvtav1 --enable-libx264 --enable-libplacebo --enable-librav1e --enable-shared
  WARNING: library configuration mismatch
  avcodec     configuration: --prefix=/usr --extra-version=7+b1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-sndio --enable-libjxl --enable-pocketsphinx --enable-librsvg --enable-libvpl --disable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --disable-libsvtav1 --enable-libx264 --enable-libplacebo --enable-librav1e --enable-shared --enable-version3 --disable-doc --disable-programs --disable-static --enable-libaribb24 --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libtesseract --enable-libvo_amrwbenc --enable-libsmbclient
  avformat    configuration: --prefix=/usr --extra-version=7+b1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-sndio --enable-libjxl --enable-pocketsphinx --enable-librsvg --enable-libvpl --disable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --disable-libsvtav1 --enable-libx264 --enable-libplacebo --enable-librav1e --enable-shared --enable-version3 --disable-doc --disable-programs --disable-static --enable-libaribb24 --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libtesseract --enable-libvo_amrwbenc --enable-libsmbclient
  avfilter    configuration: --prefix=/usr --extra-version=7+b1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-sndio --enable-libjxl --enable-pocketsphinx --enable-librsvg --enable-libvpl --disable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --disable-libsvtav1 --enable-libx264 --enable-libplacebo --enable-librav1e --enable-shared --enable-version3 --disable-doc --disable-programs --disable-static --enable-libaribb24 --enable-libopencore_amrnb --enable-libopencore_amrwb --enable-libtesseract --enable-libvo_amrwbenc --enable-libsmbclient
  libavutil      58.  2.100 / 58.  2.100
  libavcodec     60.  3.100 / 60.  3.100
  libavformat    60.  3.100 / 60.  3.100
  libavdevice    60.  1.100 / 60.  1.100
  libavfilter     9.  3.100 /  9.  3.100
  libswscale      7.  1.100 /  7.  1.100
  libswresample   4. 10.100 /  4. 10.100
  libpostproc    57.  1.100 / 57.  1.100
bdj.c:614: libbluray-j2se-1.3.4.jar not found.
bdj.c:801: BD-J check: Failed to load libbluray.jar
bdj.c:614: libbluray-j2se-1.3.4.jar not found.
bdj.c:801: BD-J check: Failed to load libbluray.jar
[bluray @ 0x560f3b3a14c0] 3 usable playlists:
[bluray @ 0x560f3b3a14c0] playlist 00004.mpls (3:12:34)
[bluray @ 0x560f3b3a14c0] playlist 00800.mpls (3:12:34)
[bluray @ 0x560f3b3a14c0] playlist 00801.mpls (3:12:34)
[bluray @ 0x560f3b3a14c0] selected 00004.mpls
[mpegts @ 0x560f3b3a0cc0] start time for stream 11 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] start time for stream 12 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] start time for stream 13 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] start time for stream 14 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] start time for stream 15 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] start time for stream 16 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] start time for stream 17 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] start time for stream 18 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] start time for stream 19 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] start time for stream 20 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] start time for stream 26 is not set in estimate_timings_from_pts
[mpegts @ 0x560f3b3a0cc0] stream 26 : no TS found at start of file, duration not set
bluray.c:299: 00062.m2ts: no timestamp for SPN 0 (got 0). clip 188955000-198097258.
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 11 (Subtitle: hdmv_pgs_subtitle (pgssub) ([144][0][0][0] / 0x0090)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 12 (Subtitle: hdmv_pgs_subtitle (pgssub) ([144][0][0][0] / 0x0090)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 13 (Subtitle: hdmv_pgs_subtitle (pgssub) ([144][0][0][0] / 0x0090)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 14 (Subtitle: hdmv_pgs_subtitle (pgssub) ([144][0][0][0] / 0x0090)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 15 (Subtitle: hdmv_pgs_subtitle (pgssub) ([144][0][0][0] / 0x0090)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 16 (Subtitle: hdmv_pgs_subtitle (pgssub) ([144][0][0][0] / 0x0090)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 17 (Subtitle: hdmv_pgs_subtitle (pgssub) ([144][0][0][0] / 0x0090)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 18 (Subtitle: hdmv_pgs_subtitle (pgssub) ([144][0][0][0] / 0x0090)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 19 (Subtitle: hdmv_pgs_subtitle (pgssub) ([144][0][0][0] / 0x0090)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 20 (Subtitle: hdmv_pgs_subtitle (pgssub) ([144][0][0][0] / 0x0090)): unspecified size
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
[mpegts @ 0x560f3b3a0cc0] Could not find codec parameters for stream 26 (Audio: ac3, 0 channels): unspecified sample format
Consider increasing the value for the 'analyzeduration' (0) and 'probesize' (5000000) options
Input #0, mpegts, from 'bluray:.':
  Duration: 03:12:27.32, start: 4199.000000, bitrate: 62841 kb/s
  Program 1 
  Stream #0:0[0x1011]: Video: hevc (Main 10) (HDMV / 0x564D4448), yuv420p10le(tv, bt2020nc/bt2020/smpte2084), 3840x2160 [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 90k tbn
  Stream #0:1[0x1100]: Audio: truehd (AC-3 / 0x332D4341), 48000 Hz, 7.1, s32 (24 bit)
  Stream #0:2[0x1100]: Audio: ac3 (AC-3 / 0x332D4341), 48000 Hz, 5.1(side), fltp, 640 kb/s
  Stream #0:3[0x1101]: Audio: dts (DTS-HD MA) ([134][0][0][0] / 0x0086), 48000 Hz, stereo, s32p (24 bit)
  Stream #0:4[0x1102]: Audio: ac3 (AC-3 / 0x332D4341), 48000 Hz, stereo, fltp, 320 kb/s
  Stream #0:5[0x1103]: Audio: ac3 (AC-3 / 0x332D4341), 48000 Hz, 5.1(side), fltp, 640 kb/s
  Stream #0:6[0x1104]: Audio: eac3 (AC-3 / 0x332D4341), 48000 Hz, 7.1, fltp, 1024 kb/s
  Stream #0:7[0x1105]: Audio: dts (DTS) ([130][0][0][0] / 0x0082), 48000 Hz, 5.1(side), fltp, 1536 kb/s
  Stream #0:8[0x1106]: Audio: dts (DTS) ([130][0][0][0] / 0x0082), 48000 Hz, 5.1(side), fltp, 1536 kb/s
  Stream #0:9[0x1107]: Audio: eac3 (AC-3 / 0x332D4341), 48000 Hz, 7.1, fltp, 1024 kb/s
  Stream #0:10[0x1108]: Audio: eac3 (AC-3 / 0x332D4341), 48000 Hz, 7.1, fltp, 1024 kb/s
  Stream #0:11[0x12a0]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090)
  Stream #0:12[0x12a1]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090)
  Stream #0:13[0x12a2]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090)
  Stream #0:14[0x12a3]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090)
  Stream #0:15[0x12a4]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090)
  Stream #0:16[0x12a5]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090)
  Stream #0:17[0x12a6]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090)
  Stream #0:18[0x12a7]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090)
  Stream #0:19[0x12a8]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090)
  Stream #0:20[0x12a9]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090)
  Stream #0:21[0x12aa]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090), 1920x1080
  Stream #0:22[0x12ab]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090), 1920x1080
  Stream #0:23[0x12ac]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090), 1920x1080
  Stream #0:24[0x12ad]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090), 1920x1080
  Stream #0:25[0x12ae]: Subtitle: hdmv_pgs_subtitle ([144][0][0][0] / 0x0090), 1920x1080
  No Program
  Stream #0:26[0x1fff]: Audio: ac3, 0 channels
```

Is that any help?

**Comment by Dudemanguy** [comment]
Not really. Can you provide a sample of the log file when you're playing this? It doesn't need to be the whole thing.

**Comment by philcerf** [comment]
Here we go, made via `mpv bd://mpls/0801 --bluray-device=. --log-file=~/mpv.log`:
[mpv.log](https://github.com/mpv-player/mpv/files/13059740/mpv.log)

**Comment by philcerf** [comment]
Also, I've just used make mkv to convert the bluray structure into a single MKV, but the problem persists.

I mean AFAIU make mkv just copies the streams 1:1 into the MKV container, so there wasn't really much difference to expect - but at least it's nothing from some bluary oddities or so.

**Comment by philcerf** [comment]
Oh and just to be sure:

You did zoom into the screenshots and notice that perfect mosaic structure of the artefacts? That seemed pretty unusual to me.

**Comment by llyyr** [comment]
Upstream kernel issue, it's fixed on 6.5.7 or 6.6.-rc5. https://gitlab.freedesktop.org/drm/intel/-/issues/9118

**Comment by philcerf** [comment]
@llyyr Sure that's the same? The report there talks about vaapi. But my issue also happens without hardware decoding.
Anyway,.. I'll try it with 6.5.7, once that lands in Debian.

**Comment by ValeZAA** [comment]
>Okay I can try it again with ffplay and with hardware decoding (once I found out how to do that ^^ … you don't happen to know?).

Sure. Apply these 2 patches https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=9895

**Comment by haasn** [comment]
> ffplay


> > Okay I can try it again with ffplay and with hardware decoding (once I found out how to do that ^^ … you don't happen to know?).
> 
> Sure. Apply these 2 patches https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=9895

You can use also `plplay --hwdec`.

**Comment by philcerf** [comment]
Hey.

I haven't tried with 6.5.7 yet (too lazy too compile it ^^), but I did try the workaround (using `--vo=gpu-next --gpu-api=vulkan`) that was given in https://gitlab.freedesktop.org/drm/intel/-/issues/9118#note_2099221 … and with that it indeed works.

I still don't get why it also fails without hardware decoding (presumably because even then it uses some of the broken functionality from the driver), but from my side, we can close this issue as duplicate of presumably #12364?

Should it persist after 6.5.7, I'd come back.

Thanks,
Philippe

**Comment by philcerf** [comment]
Just to confirm, it's indeed fixed. Tested with 6.5.8.
